app/src/test/java/org/simple/clinic/enterotp/EnterOtpUpdateTest.kt

```diff
@@ -0,0 +1,33 @@
+package org.simple.clinic.enterotp
+
+import com.spotify.mobius.test.NextMatchers.hasModel
+import com.spotify.mobius.test.NextMatchers.hasEffects
+import com.spotify.mobius.test.UpdateSpec
+import com.spotify.mobius.test.UpdateSpec.assertThatNext
+import org.junit.Test
+import org.simple.clinic.TestData
+import org.simple.clinic.login.LoginResult.NetworkError
+import java.util.UUID
+
+class EnterOtpUpdateTest {
+  @Test
+  fun `when the login request is completed and is unsuccessful, then finish logging in and show error`() {
```

Comment:
For what it's worth, now that we have this test here, we should delete the tests from the regression test suite that were verifying this behaviour.

We can do this in another PR, I'm going to approve this for now.